src/main.h

```diff
@@ -511,9 +520,6 @@ class CVerifyDB {
     bool VerifyDB(const CChainParams& chainparams, CCoinsView *coinsview, int nCheckLevel, int nCheckDepth);
 };
 
-/** Find the last common block between the parameter chain and a locator. */
-CBlockIndex* FindForkInGlobalIndex(const CChain& chain, const CBlockLocator& locator);
```

Comment:
Please don't move validation related methods to the network layer.

This one specifically is also used in init.cpp

Reply:
woops, will move back

Reply:
btw, I got curious to see in what state bitcoin-core was in this respect and they got rid of main.cpp completely. they moved most of this stuff to [`net_processing.cpp`](https://github.com/bitcoin/bitcoin/blob/master/src/net_processing.cpp), for some reason, perhaps because it was 2am, I thought I had seen that method there.

What do you think about moving ProcessMessages and SendMessages and their utility methods from main.cpp into here?

Reply:
In another branch I have a refactor for the validation code, which is a super sensitive, and the result would be another couple thousand of lines of code moved out of main.cpp, so I agree with your goal.

I have not looked at the impact of moving those methods. So without knowing the full extend I can say that in general I agree.
I would advice to take it slow. Make things work and compile and all the tests pass between movings of code. Makes things easier in the end.